src/cas/pyodide-cas.ts

```diff
@@ -111,25 +111,87 @@ function usePythonConverter() {
     ['EqualEqual', () => 'sympy.Eq'],
+    ['Parentheses', () => ''],
+
+    ['Abs', () => 'sympy.Abs'],
+
+    ['Sin', () => 'sympy.sin'],
+    ['Cos', () => 'sympy.cos'],
+    ['Tan', () => 'sympy.tan'],
+
+    ['Sec', () => 'sympy.sec'],
+    ['Csc', () => 'sympy.csc'],
+    ['Cot', () => 'sympy.cot'],
+
+    ['Arcsin', () => 'sympy.asin'],
+    ['Arccos', () => 'sympy.acos'],
+    ['Arctan', () => 'sympy.atan'],
+
+    ['Asec', () => 'sympy.asec'],
+    ['Acsc', () => 'sympy.acsc'],
+    ['Atan', () => 'sympy.atan'],
+
+    ['Sinh', () => 'sympy.sinh'],
+    ['Cosh', () => 'sympy.cosh'],
+    ['Tanh', () => 'sympy.tanh'],
+
+    ['Sech', () => 'sympy.sech'],
+    ['Csch', () => 'sympy.csch'],
+    ['Coth', () => 'sympy.coth'],
+
+    ['Arsinh', () => 'sympy.asinh'],
+    ['Arcosh', () => 'sympy.acosh'],
+    ['Artanh', () => 'sympy.atanh'],
+
+    ['Asech', () => 'sympy.asech'],
+    ['Acsch', () => 'sympy.acsch'],
+    ['Acoth', () => 'sympy.acoth'],
+
+    ['Pi', () => 'sympy.pi'],
+    ['ImaginaryUnit', () => 'sympy.I'], // TODO: test
+    ['ExponentialE', () => 'sympy.E'], // TODO: test
+    ['GoldenRatio', () => 'sympy.GoldenRatio'], // TODO: test
+    ['EulerGamma', () => 'sympy.EulerGamma'], // TODO: test
+
+    // TODO: MathLive latex serial/parse destroys ln,log, etc? in ExpressionElement.vue line 78
+    // ['log', () => 'sympy.log'],
   ])
 
+  const KnownLatexFunctions = {
+    '\\sin': 'sympy.sin',
+  }
+
   // TODO: Options (rational numbers)
   function expressionToPython(expression: any): string {
     if (Array.isArray(expression)) {
+      // Handle Functions (Add, Power, Sin, etc.)
       expression = expression.slice() // Make a copy so that we can safely modify the expression
       const functionName = expression[0]
-      const toPython = MathJsonToSympy.get(functionName)
-      let pythonFunctionName = toPython ? toPython(expression) : functionName
+      const functionInPython = MathJsonToSympy.get(functionName)
+      let pythonFunctionName = functionInPython ? functionInPython(expression) : functionName
 
       const parameters = []
       for (let i = 1; i < expression.length; i++) {
         parameters.push(expressionToPython(expression[i]))
       }
       return `${pythonFunctionName}(${parameters.join(',')})`
     } else if (typeof expression === 'string') {
-      return encodeName(expression)
+      const constantInPython = MathJsonToSympy.get(expression)
+      // TODO: Handle Units here?
+      if (constantInPython) {
+        // Handle Constants (pi, e, etc.)
+        return constantInPython([])
+      } else {
+        // Handle Variables
+        return encodeName(expression)
+      }
     } else if (typeof expression === 'number') {
-      return `sympy.Float(${expression})`
+      // Handle Numbers
+      if (Number.isInteger(expression)) {
+        return `sympy.Integer(${expression})`
+      } else {
+        return `sympy.Float(${expression})`
```

Comment:
In the spirit of https://github.com/stefnotch/quantum-sheet/issues/28 , please try to change this to `sympy.Rational()`. If that works, then wohoo!

We should probably also make it configureable whether to use `sympy.Rational` or `sympy.Float`

Reply:
I have changed it to Rational, also included cases of `{'num': X}` too